src/test/java/org/broadinstitute/consent/http/db/DataSetDAOTest.java

```diff
@@ -158,6 +163,31 @@ public void testFindDatasetPropertiesByDatasetId() {
         assertEquals(properties.size(), 1);
     }
 
+    @Test
+    public void testUpdateDataset() {
+        DataSet d = createDataset();
+        Timestamp now = new Timestamp(new Date().getTime());
+        dataSetDAO.updateDatasetUpdateUserAndDate(d.getDataSetId(), now, d.getCreateUserId());
+        DataSet updated = dataSetDAO.findDataSetById(d.getDataSetId());
+        assertEquals(updated.getUpdateDate(), now);
+    }
+
+    @Test
+    public void testUpdateDatasetProperty() {
+        DataSet d = createDataset();
+        Set<DataSetProperty> properties = dataSetDAO.findDatasetPropertiesByDatasetId(d.getDataSetId());
+        DataSetProperty originalProperty = properties.stream().collect(Collectors.toList()).get(0);
+        DataSetProperty newProperty = new DataSetProperty(d.getDataSetId(), 1, "Updated Value", new Date());
+        List<DataSetProperty> updatedProperties = new ArrayList<>();
+        updatedProperties.add(newProperty);
+        dataSetDAO.updateDatasetProperty(d.getDataSetId(), updatedProperties.get(0).getPropertyKey(), updatedProperties.get(0).getPropertyValue());
+        Set<DataSetProperty> returnedProperties = dataSetDAO.findDatasetPropertiesByDatasetId(d.getDataSetId());
+        DataSetProperty returnedProperty = returnedProperties.stream().collect(Collectors.toList()).get(0);
+        assertEquals(originalProperty.getPropertyKey(), returnedProperty.getPropertyKey());
+        assertEquals(originalProperty.getPropertyId(), returnedProperty.getPropertyId());
+        assertNotEquals(originalProperty.getPropertyValue(), returnedProperty.getPropertyValue());
+    }
```

Comment:
Some more test ideas: 
* Update with multiple changed props
* Update with one changed prop and one unchanged prop
* Update with multiple unchanged props
* If we do tackle removed props, test that too.

Reply:
Gotcha, thanks!